allennlp/tests/training/optimizer_test.py

```diff
@@ -68,7 +68,7 @@ def test_optimizer_parameter_groups(self):
 
     def test_parameter_type_inference(self):
         # Should work ok even with lr as a string
```

Comment:
This appears to be a regression, but perhaps it's no longer relevant. @matt-gardner, does your recent work make this unnecessary? If so, this test should just be deleted.

Reply:
Argh, I didn't even read the comment above. Clearly something has to change.

Reply:
I think the ability to handle this was added because of `std.extvar` substitution (or an earlier version of that) gave strings instead of floats in the json.  I'm ok removing it, but we may need to reconsider this at some point if we get people complaining about things not working with environment variables.  Should be easy to handle this again if we need to, though.

Reply:
Ah, right. That does seem useful. But we can suggest people use `std.parseJson` (there is no std.parseFloat, see https://github.com/google/jsonnet/issues/759) as an alternative.

Reply:
To be clear, I'm really not sure if this is still necessary or not; I think we did this before we moved to jsonnet, and it may no longer be necessary.

Reply:
For context, this change was added about a year ago when jsonnet was in place. See https://github.com/allenai/allennlp/pull/2324. (I hold no strong opinions on the functionality either way.)